python/ql/src/Security/CWE-327/TlsLibraryModel.qll

```diff
@@ -0,0 +1,134 @@
+import python
+import semmle.python.ApiGraphs
+import Ssl
+import PyOpenSSL
+
+/**
+ * A specific protocol version.
+ * We use this to identify a protocol.
+ */
+class ProtocolVersion extends string {
+  ProtocolVersion() { this in ["SSLv2", "SSLv3", "TLSv1", "TLSv1_1", "TLSv1_2", "TLSv1_3"] }
+
+  predicate lessThan(ProtocolVersion version) {
```

Comment:
```suggestion
  /** Holds if this protocol version predates `version` */
  predicate lessThan(ProtocolVersion version) {
```

Hmm :thinking: I'm not sure why you don't model `SSLv2` < `TLSv1` though :question:

Reply:
This is for use with `minimum_version`, so it is not a total order, rather stratified on protocols.

Reply:
But, if you allow `SSLv3` as minimum version, won't `TLSv1`, `TLSv1_1`, `TLSv1_2`, and `TLSv1_3` also be allowed? (that is at least my interpretation from reading the documentation). So SSLv2 < SSLv3 < TLSv1 < TLSv1_1 < TLSv1_2 < TLSv1_3

Reply:
What are you reading? I naturally see TLS and SSL as different protocols with unrelated versions. If we should compare across protocols, I think we need to include a reference in a comment in the code.